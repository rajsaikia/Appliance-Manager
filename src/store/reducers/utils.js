import { cloneDeep } from 'lodash';

export const filterRooms = rooms => {
	return rooms.filter(res => res.name);
};

export const updateAppliances = (appliances, source) => {
	const newUpdateAppliances = cloneDeep(appliances);
	const orderAt = newUpdateAppliances.findIndex(o => o.id === source.id);
	const app = newUpdateAppliances.filter(res => res.id === source.id);
	const keynames = Object.keys(source);
	app[0].nickname = source.nickname;
	app[0].properties.map(res =>
		keynames.includes(res.name) ? (res.value = source[res.name]) : null
	);
	newUpdateAppliances[orderAt] = app[0];
	return newUpdateAppliances;
};

export const addNewAppliances = (appliances, payload) => {
	const newUpdateAppliances = cloneDeep(appliances);
	const data = {
		...payload,
		...{
			type: 'New Appliances',
			nickname: 'NA',
			properties: [
				{
					name: 'power',
					type: 'toggle',
					value: true,
				},
			],
		},
	};
	newUpdateAppliances.push(data);
	return newUpdateAppliances;
};

export const addAppliancesToRoom = (rooms, selectedRoom, payload) => {
	const newUpdateRooms = cloneDeep(rooms);
	newUpdateRooms
		.filter(res => res.id === selectedRoom)[0]
		.applianceIds.push(payload.id);
	return newUpdateRooms;
};
export const AddAppliancesIndex = (appliancesIndex, payload) => {
	const newUpdateRooms = cloneDeep(appliancesIndex);
	newUpdateRooms.push(payload.id);
	return newUpdateRooms;
};
