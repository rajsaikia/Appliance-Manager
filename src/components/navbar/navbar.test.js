import React from 'react';
import { render } from 'jest/utils';
import '@testing-library/jest-dom/extend-expect';
import Navbar from './index';

describe('<Checkbox /> - Component', () => {
	let container = null;

	const Component = () => {
		return <Navbar />;
	};
	beforeEach(async () => {
		container = render(<Component />);
	});

	afterEach(() => {
		container.unmount();
	});
	it('renders correctly', async () => {
		expect(container.asFragment()).toMatchSnapshot();
	});
	it('it should contain the text ApplianceManager', async () => {
		expect(container.getByText('ApplianceManager')).toBeInTheDocument();
	});
});
