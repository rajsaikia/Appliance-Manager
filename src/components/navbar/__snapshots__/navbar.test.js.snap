// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> - Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      class="navbar navbarSection"
    >
      <span
        class="navbar-brand mb-0 h1"
      >
        ApplianceManager
      </span>
    </nav>
  </div>
</DocumentFragment>
`;
